library/sha512.c

```diff
@@ -22,8 +22,36 @@
  *  http://csrc.nist.gov/publications/fips/fips180-2/fips180-2.pdf
  */
 
+#if defined(__aarch64__) && !defined(__ARM_FEATURE_SHA512) && \
+    defined(__clang__) &&  __clang_major__ < 18 && \
+    __clang_major__ >= 13 && __clang_minor__ > 0 && __clang_patchlevel__ > 0
```

Comment:
Do we want to do this (i.e. mess with the compilation target) only if we have one of `MBEDTLS_SHA512_USE_A64_CRYPTO_xxx`?

Reply:
At this point, `MBEDTLS_SHA512_USE_A64_CRYPTO_xxx` are not defined. They are defined by `common.h` .  
 `MBEDTLS_SHA512_USE_A64_CRYPTO_xxx`  can not be used here.